luna-manager/src/Luna/Manager/Archive.hs

```
@@ -58,16 +51,19 @@ unpackingException t e = toException $ UnpackingException t e
 unpack :: UnpackContext m => Double -> Text.Text -> FilePath -> m FilePath
 unpack totalProgress progressFieldName file = do
     Logger.info $ "Unpacking archive: " <> plainTextPath file
-    ext          <- tryJust (extensionError file) $ extension file
+    let ext = fromMaybe "" $ extension file
     case currentHost of
         Windows -> case ext of
             "zip" -> unzipFileWindows file
             "gz"  -> untarWin totalProgress progressFieldName file
+            "exe" -> return file
```

Please add some sensible catch-all clauses here, so that if anybody makes a mistake and types e.g. "eexe" or "zipp" as an extension, it won't fail on an incomplete pattern match